### PR TITLE
Flatten nested object tables

### PR DESCRIPTION
**Pull request by kustosz**
This improves the way our nested tables visualizations work. If you have a uniform list of objects like:
```
[ {foo: {foo1: ..., foo2: ...}, {bar: {bar1: ..., bar2: ...}}
, {foo: {foo1: ..., foo2: ...}, {bar: {bar1: ..., bar2: ...}}
, ...]
```
The way it used to work was you'd get two columns (`foo` and `bar`) and then each cell would become a tiny two-column table. This in turn resulted in bloating the table (the nested headers would appear for each table cell).

With this change, the structure gets visualized as a 4-column table, with `foo.foo1`, `foo.foo2`, `bar.bar1` and `bar.bar2` columns. No more redundancy in the structure!

### CODE REVIEW

File: luna-studio/atom/lib/visualizers/base/table/table.js [hunk 1]
```diff
@@ -87,6 +115,7 @@
     } else if (data instanceof Object) {
       return "<td>" + genRowObjectTable(data, level + 1) + "</td>";
     } else {
+      if (data === undefined || data === null) data = "";
```
Comment (piotrMocz) [inline luna-studio/atom/lib/visualizers/base/table/table.js]: Or: `var res = (data || "").toString();`

File: luna-studio/atom/lib/visualizers/base/table/table.js [hunk 2]
```diff
@@ -9,6 +9,33 @@
...
+
+  var generateNestings = function (data, key) {
+    var first = getAtNestedKey(data[0], key);
+    if (!(first instanceof Object)) return [key];
+    var firstKeys = Object.keys(first);
+    var isNestable = data.every(obj => hasExactlyKeys(firstKeys, getAtNestedKey(obj, key)));
+    if (isNestable) {
+      var withNests = firstKeys.map(k => key.concat([k]));
+      var furtherNestings = withNests.map(k => generateNestings(data, k));
+      return [].concat.apply([], furtherNestings);
```
Comment (piotrMocz) [inline luna-studio/atom/lib/visualizers/base/table/table.js]: I'm not sure I like this syntax very much. Confusing for future readers of the code, you have to explicitly know what it does. I wouldn't bet my head on it, but isn't it the same as:
```
return furtherNestings(reduce((x, y) => x.concat(y));
```
?
If yes, I would change it for the reader's convenience :)

File: luna-studio/atom/lib/visualizers/base/table/table.js [hunk 3]
```diff
@@ -9,6 +9,33 @@
     return Object.keys(obj).length == keys.length && keys.every(k => obj.hasOwnProperty(k));
   };
 
+  var getAtNestedKey = function (data, key) {
+    var res = data;
+    key.forEach(function (k) { res = res[k]; });
```
Comment (piotrMocz) [inline luna-studio/atom/lib/visualizers/base/table/table.js]: You can actually use the arrow syntax with statements, like: `key.forEach(k => { res = res[k]; });`

File: luna-studio/atom/lib/visualizers/base/table/table.js (hunk 2 above)
Comment (kustosz) [reply luna-studio/atom/lib/visualizers/base/table/table.js]: Not sure, isn't that slower somehow? Like multiple copies of the intermediate arrays being created?

File: luna-studio/atom/lib/visualizers/base/table/table.js (hunk 1 above)
Comment (kustosz) [reply luna-studio/atom/lib/visualizers/base/table/table.js]: Nope, I'd still like to have my `false`s and `0`s displayed like they are supposed to be.

File: luna-studio/atom/lib/visualizers/base/table/table.js (hunk 2 above)
Comment (piotrMocz) [reply luna-studio/atom/lib/visualizers/base/table/table.js]: I am not sure, I like to believe in reasonable compiler optimizations. You can let it be, but maybe write a comment what it does? I needed a few minutes to figure that out :D

File: luna-studio/atom/lib/visualizers/base/table/table.js (hunk 1 above)
Comment (piotrMocz) [reply luna-studio/atom/lib/visualizers/base/table/table.js]: Fair point, sir.

**Pull request merged**